### ISSUE TITLE
Cannot connect to iCloud CalDAV & CardDAV (authentication works)

### ISSUE THREAD

**Issue by KraIstGerecht** [jobisoft/TbSync#264]
## Your environment

TbSync version: Same issue in stable 1.7 and beta 1.7.6
Thunderbird version: 60.8.0

[x] Yes, I have installed the latest available (beta) version from 
https://github.com/jobisoft/TbSync/releases 
and my issue is not yet fixed, I can still reproduce it.

## Expected behavior
Connection to iCloud CalDAV and CardDAV can be established.

## Actual behavior
TbSync has been running like a charm since the day I discovered it a few months ago (Thanks John!). But since a few weeks I've been having a weird issue where sometimes TbSync would show an error saying that it can't find a CalDav or CardDav server for my iCloud account, and then it would work again after either restarting Thunderbird, setting up a new account the same way the old one was set up or simply hammering the "Sync now" button a few times.

It has apparently stopped working completely about 3 days ago, I hadn't done any changes to the way things are set up. When I try setting up a new account now the authentication with iCloud will work fine, but even with a correct app-specific password it always says it can't connect to the server. This doesn't produce any entries in the error protocol. Here's a list of what I've tried to do in order to work around the issue, without success:

-made sure my internet connection is fine.
-checked I had the current versions of TB and TbSync.
-re-installed the stable version of TbSync.
-re-installed the CalDAV & CardDAV provider separately.
-installed the beta channel version of TbSync.
-had new app-specific passwords generated for my Apple ID and used those.
-checked whether the iCloud service was down (since it's been like this for days, I deem this unlikely)
-checked the TbSync timeout for server connections (is at the default 90000)

I really can't tell whether there's something wrong with TbSync itself, with the cloud service or whether the problem is on my end and I am making some kind of stupid mistake.

I've sent the debug log via e-mail too.

## Steps to reproduce
Attempting to set up a new iCloud CalDAV & CardDAV account.

To help resolving your issue, enable debug logging (TbSync Account Manager -> Help) and send me the debug.log via e-mail (use the title of your issue as subject of the email).

**Comment by jobisoft** [comment]
Apple has messed up their caldav server and since TbSync req. both (caldav and carddav) to work, it currently fails, even though contact sync would work. 

The caldav server returns this on the initial request (set extensions.tbsync.log.userdatalevel to 9 and activate debug log to see that):

```
URL:
https://caldav.icloud.com/ (PROPFIND)

Request:
<d:propfind xmlns:d="DAV:"><d:prop><d:current-user-principal /></d:prop></d:propfind>

Response:
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<multistatus xmlns="DAV:">
    <response xmlns="DAV:">
        <href>/</href>
            <propstat>
                <prop>
                    <current-user-principal xmlns="DAV:">
                        <href xmlns="DAV:">/1776546672/principal/</href>
                    </current-user-principal>
                </prop>
                <status>HTTP/1.1 404 Not Found</status>
            </propstat>
    </response>
</multistatus>
```

Even though it returns the requested info, it sets a 404 error status which causes TbSync to discard the URL.

Looking into it.

**Comment by KraIstGerecht** [comment]
Right, thanks for the quick heads up! At least now I know it wasn't anything I screwed up myself. I'll wait and see what happens.

**Comment by jobisoft** [comment]
I have a fix. Release on Monday.

**Comment by KraIstGerecht** [comment]
Thanks a bunch for the fix.

**Comment by jobisoft** [comment]
Here we go.

The official version on ATN is pending review. I think it will be released by the Thunderbird team tomorrow. It will be version 0.15.0.1 of the CalDAV & CardDAV provider.

If you are on the beta release channel, just open your AddonManager and search for updates as described [here](https://github.com/jobisoft/TbSync/wiki/Get-the-latest-TbSync-version#geting-updates-via-the-beta-release-channel). You will get the fixed version 0.15.14 of the CalDAV & CardDAV provider.

Furthermore: Today I also released a beta version of TbSync for the next ESR of Thunderbird: TB68. More information about the two beta release channels (TB60 and TB68) can be found here:
https://tbsync.jobisoft.de

This has been a lot of work. If you like TbSync, please consider a donation, so I can continue to provide support.

[![](https://www.paypalobjects.com/en_US/DK/i/btn/btn_donateCC_LG.gif)](https://www.paypal.me/johnbieling)

**Comment by jobisoft** [comment]
Thunderbird team has released the fixed version to the public.
